src/api/OpenWith.js

```diff
@@ -32,91 +34,126 @@ class OpenWith extends Base {
-                this.fetchAppIntegrations(formattedOpenWithIntegrations, successCallback, errorCallback);
+                this.fetchAppIntegrations(openWithIntegrations, successCallback, errorCallback);
             },
             errorCallback
         });
     }
 
-    /**
-     * Formats data conveniently for the client
-     *
-     * @param {OpenWithIntegrations} openWithIntegrations - The available Open With integrations
-     * @return {Array<OpenWithIntegrationItem>} formatted open with integrations
-     */
-    formatOpenWithData(openWithIntegrations: OpenWithIntegrations): Array<any> {
-        // Eventually, a default integration will be returned outside of the items array,
-        // which should be inserted into this list of integrations.
-
-        const { items } = openWithIntegrations;
-        // Fix the API response by converting the app integration ID to a string
-        const integrations = items.map((item) => ({
-            ...item,
-            app_integration: {
-                id: item.app_integration.id.toString(),
-                type: 'app_integration'
-            }
-        }));
-
-        // Sort integrations by display_order
-        return integrations.sort(
-            (integrationA: OpenWithIntegrationItem, integrationB: OpenWithIntegrationItem) =>
-                integrationA.display_order - integrationB.display_order
-        );
-    }
-
     /**
      * Fetch app integrations info needed to render.
      *
      * @param {Array<OpenWithIntegrationItem>} openWithIntegrations - The available Open With integrations
      * @return {void}
      */
     fetchAppIntegrations = async (
-        openWithIntegrations: Array<OpenWithIntegrationItem>,
+        openWithIntegrations: OpenWithAPI,
         successCallback: Function,
         errorCallback: Function
     ) => {
+        const items = this.addDefaultToOpenWithItems(openWithIntegrations);
         this.appIntegrationsAPI = new AppIntegrationsAPI(this.options);
 
-        const appIntegrationPromises = openWithIntegrations.map((integrationItem: OpenWithIntegrationItem) => {
+        const appIntegrationPromises = items.map((integrationItem: Object) => {
             const { app_integration: { id } } = integrationItem;
             return this.appIntegrationsAPI.fetchAppIntegrationsPromise(id);
         });
 
         try {
-            const appIntegrations: Array<AppIntegrationItem> = await Promise.all(appIntegrationPromises);
-            const completedOpenWithIntegrations = this.completeOpenWithIntegrationData(
-                openWithIntegrations,
-                appIntegrations
-            );
-            successCallback(completedOpenWithIntegrations);
+            const appIntegrations: Array<AppIntegrationAPIItem> = await Promise.all(appIntegrationPromises);
+            const completedOpenWithIntegrations = this.completeOpenWithIntegrationData(items, appIntegrations);
```

Comment:
What do you think about combining ```completeOpenWithIntegrationData``` with ```formatOpenWithData```? You are mapping over the object in both places